Supplemental Material of the article "Within-host rhinovirus evolution in upper and lower respiratory tract highlights capsid variability and mutation-independent compartmentalization"

**Figure_Generation.Rmd**
Scripts to generate analysis and figures for most of paper: Figure 1, 2, S3, S5, S7, S8

**1_AYM_All_Dataframes**
Script to combine all dataframes to generate Table S2

**Alignments**
Fasta allignments done through HHpred of RV A, B, and C and epitopes

**Deep_Mutation**
Deep mutational script for alligning fastas and mapping deep mutational 4GB3 to 1AYMN capsid.  

**Epitopes**
Contains dataframes and maps epitope positions to each RV capsid.

**Interpentamer_Sites**
Maps interpentameter positions to each RV capsid.

**MAF_vs_SASA**
Maps surface area accessibility to RV A to generate surface area analysis and Figure 3. All pymol files are contained in directory.

**mNGS_PCR**
Script to combine and reformate data to generate figure S1.

**RV_LAVA_DATAFRAMES**
Combines all lava output dataframes together.

**Structure_Protein**
Annotates structure of iSNVs within the capside using EMBOSS

**Surface_Exposure**
Annotates position of iSNVs within the capside using viperDB

**viperdb_info**
viperDB dataframes

**Microbiome_analysis**
Microbiome characterization in upper and lower respiratory tract samples from immunocompromised patients during HRV infection to evaluate cross-contamination. 

**Revica-Coinfection**
Detailed summary of the metagenomic analysis performed with Revica (https://github.com/greninger-lab/revica). 
