Supplemental Material of the article "Within-host rhinovirus evolution in upper and lower respiratory tract highlights capsid variability and mutation-independent compartmentalization"

**Revica-Coinfection**
Detailed summary of the metagenomic analysis performed with Revica (https://github.com/greninger-lab/revica). 

**LAVA_dataframes**
LAVA pipeline (https://github.com/greninger-lab/lava/tree/Lava_Slippage-Patch) results for the detection of intra-host single nucleotide variant (iSNV) with a minimum allele frequency (MAF) of 10% in the positions with a minimum sequence depth of coverage of 30X

**Comparison_PCR_mNGS**
Comparatison of iSNV detection with LAVA in samples sequenced by mNGS and amplicon-based.

**MAF_vs_SolventAccesibilitySolventArea**
Study of the intra-host viral allele frequency (AF) and the solvent accessibility surface area (SASA) for each iSNV located in the capsid proteins of HRV-A samples.

**Microbiome_analysis**
Microbiome characterization in upper and lower respiratory tract samples from immunocompromised patients during HRV infection to evaluate cross-contamination. 

**Deep_Mutational_Scanning_analysis**
study of the mutational fitness effect in a picornavirus capsid and the presence of intra-host single nucleotide variant (iSNV).
